src/main/java/sirius/biz/storage/layer3/DeleteFilesJob.java

```diff
@@ -0,0 +1,353 @@
+     * @return <tt>true</tt> if the file has been deleted, <tt>false</tt> otherwise
+     */
+    private boolean deleteVirtualFile(VirtualFile file) {
+        boolean isDirectory = file.isDirectory();
+        try {
+            if (!file.canDelete()) {
+                String messageKey =
+                        isDirectory ? "DeleteFilesJob.directory.cannotDelete" : "DeleteFilesJob.file.cannotDelete";
+                process.log(ProcessLog.warn()
+                                      .withNLSKey(messageKey)
+                                      .withContext(PATH_KEY, file.path())
+                                      .withMessageType("$DeleteFilesJob.warn.cannotDelete"));
+                if (isDirectory) {
+                    incrementDirectorySkipped();
+                } else {
+                    incrementFileSkipped();
+                }
+                return false;
+            }
+
+            if (!simulation) {
+                file.delete();
+            }
+            String messageKey = isDirectory ? "DeleteFilesJob.directory.deleted" : "DeleteFilesJob.file.deleted";
+            process.log(ProcessLog.info().withNLSKey(messageKey).withContext(PATH_KEY, file.path()));
+            process.incCounter(isDirectory ?
+                               "DeleteFilesJob.count.directoriesDeleted" :
+                               "DeleteFilesJob.count.filesDeleted");
+        } catch (HandledException exception) {
+            process.log(ProcessLog.error().withMessage(exception.getMessage()));
+            return false;
+        }
+
+        return true;
+    }
+
+    private void incrementFileSkipped() {
+        process.incCounter("DeleteFilesJob.count.filesSkipped");
+    }
+
+    private void incrementDirectorySkipped() {
+        process.incCounter("DeleteFilesJob.count.directoriesSkipped");
+    }
+
+    /**
+     * Defines a factory to create new instances of {@link DeleteFilesJob}.
+     */
+    @Register(framework = StorageUtils.FRAMEWORK_STORAGE)
+    @Permission(VirtualFileSystemController.PERMISSION_VIEW_FILES)
+    public static class DeleteFilesJobFactory extends DefaultBatchProcessFactory {
+
+        @Override
+        protected void collectParameters(Consumer<Parameter<?>> parameterCollector) {
+            parameterCollector.accept(SOURCE_PATH_PARAMETER);
+            parameterCollector.accept(DELETE_RECURSIVE_PARAMETER);
+            parameterCollector.accept(DELETE_EMPTY_DIRECTORIES_PARAMETER);
+            parameterCollector.accept(PATH_FILTER_PARAMETER);
+            parameterCollector.accept(LAST_MODIFIED_BEFORE_PARAMETER);
+            parameterCollector.accept(ONLY_UNUSED_PARAMETER);
+            parameterCollector.accept(SIMULATION_PARAMETER);
+        }
+
+        @Override
+        public String getCategory() {
+            return StandardCategories.MISC;
+        }
+
+        @Override
+        public String getIcon() {
+            return "fa fa-trash";
```

Comment:
Bitte Font Awesome 6 verwenden.

```suggestion
            return "fa-solid fa-trash";
```

Reply:
No, we did not "mess up" but decided to use the fallback mechanism _for existing code_ that FA 6 provides (for now). The idea was to upgrade the respective places as we go. In fact, I am currently replacing these classes while working on the Bootstrap upgrade. However, all that is no justification to still use FA 5 classes in new code. 😉